src/app/static/src/app/components/outputTable/NewFilterSelect.vue

```diff
@@ -0,0 +1,104 @@
+            <span v-if="labelTooltip"
+                v-tooltip="{
+                        content: `<dl>${labelTooltip}</dl>`,
+                        // Keep the tooltip open when mouse over the popup
+                        popperTriggers: ['hover'],
+                    }">
+                <vue-feather type="help-circle" size="20" class="ml-1"></vue-feather>
+            </span>
+        </div>
+        <single-select v-if="!multiple"
+                       :options="options"
+                       :model-value="treeselectValue"
+                       @update:model-value="input"
+                       :placeholder="placeholder"/>
+        <multi-select v-if="multiple"
+                      :options="options"
+                      :model-value="treeselectValue"
+                      @update:model-value="input"
+                      :placeholder="placeholder"/>
+    </div>
+</template>
+
+<script lang="ts">
+    import i18next from "i18next";
+    import {flattenOptions, mapStateProp} from "../../utils";
+    import {RootState} from "../../root";
+    import {Language} from "../../store/translations/locales";
+    import {FilterOption} from "../../generated";
+    import VueFeather from "vue-feather";
+    import { PropType, defineComponent } from "vue";
+    import { SingleSelect, MultiSelect } from "@reside-ic/vue-nested-multiselect";
+
+    export default defineComponent({
+        name: "FilterSelect",
+        props: {
+            multiple: {
+                type: Boolean,
+                required: false
+            },
+            label: {
+                type: String,
+                required: true
+            },
+            disabled: {
+                type: Boolean,
+                required: false,
+                default: false
+            },
+            options: {
+                type: Array as PropType<FilterOption[]>,
+                required: true
+            },
+            value: {
+                type: [Array, String] as PropType<string | string[]>,
+                required: true
+            }
+        },
+        data() {
+            const idArray = Array.isArray(this.value) ? this.value : [this.value];
+            const flatOptions = Object.values(flattenOptions(this.options));
+            const selected = flatOptions.filter((o: FilterOption) => idArray.includes(o.id));
+            return {
+                selectedOptions: selected
+            }
+        },
+        computed: {
+            treeselectValue() {
+                return this.disabled ? null : this.multiple ? this.value : this.value[0];
+            },
+            currentLanguage: mapStateProp<RootState, Language>(null,
+                (state: RootState) => state.language),
+            placeholder() {
+                const key = this.disabled ? "notUsed" : "select";
+                return i18next.t(key, this.currentLanguage)
+            },
```

Comment:
This over a v-translate? Or is there some reason that doesn't work in this context?

Reply:
i think v-translate would probably work, not sure why, just copied and pasted the old component and changed the bits i needed, will focus more on cleanup when i do the big plots cleanup project!